code/modules/mob/mob.dm

```diff
@@ -1379,6 +1379,6 @@
 /mob/proc/set_stat(new_stat)
 	if(new_stat == stat)
 		return
-	SEND_SIGNAL(src, COMSIG_MOB_STATCHANGE, new_stat)
 	. = stat
 	stat = new_stat
+	SEND_SIGNAL(src, COMSIG_MOB_STATCHANGE, .)
```

Comment:
Moving the signal to after the stat change prevents recursion when the signaled procs do something to call `set_stat`.